service/src/IOUringFallback.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#include "IOUringFallback.hpp"
+
+#include <unistd.h>
+
+#include "geopm/Helper.hpp"
+
+#include <utility>
+
+namespace geopm
+{
+    IOUringFallback::IOUringFallback(unsigned entries)
+        : m_operations()
+    {
+        m_operations.reserve(entries);
+    }
+
+    IOUringFallback::~IOUringFallback()
+    {
+    }
+
+    void IOUringFallback::submit()
+    {
+        for (const auto &operation : m_operations) {
+            errno = 0;
+            int ret = operation.second();
+            if (ret < 0) {
+                ret = -errno;
+            }
```

Comment:
```suggestion
        errno = 0;
        for (const auto &operation : m_operations) {
            int ret = operation.second();
            if (ret < 0) {
                ret = -errno;
            }
            errno = 0;

```

Reply:
Make sure to leave errno zero when the loop completes.  Nice use of bind though! Very clean.